frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow.tsx

```diff
@@ -0,0 +1,315 @@
+  formInput: any;
+  isAdmin: boolean;
+} & DashboardHeaderButtonRowProps;
+
+type ButtonConfig = {
+  component: (props: HeaderButtonProps) => ReactNode;
+  enabled: (props: HeaderButtonProps) => boolean;
+};
+
+const buttonConfigs: Record<ButtonKey, ButtonConfig> = {
+  [BUTTON_CONFIG.ADD_QUESTION]: {
+    component: () => <AddQuestionButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_HEADING_OR_TEXT]: {
+    component: () => <AddHeadingOrTextButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_LINK_CARD]: {
+    component: () => <AddLinkCardButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_SECTION]: {
+    component: () => <AddSectionButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_TEMPORAL_UNIT]: {
+    component: () => <AddTemporalUnitButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_FILTER_PARAMETER]: {
+    component: () => <AddFilterParameterButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_ACTION_ELEMENT]: {
+    component: () => <AddActionElementButton />,
+    enabled: ({ isEditing, canEdit, hasModelActionsEnabled }) =>
+      isEditing && canEdit && hasModelActionsEnabled,
+  },
+  [BUTTON_CONFIG.EXTRA_EDIT_BUTTONS_MENU]: {
+    component: () => <ExtraEditButtonsMenu />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.COPY_ANALYTICS_DASHBOARD]: {
+    component: () => <CopyAnalyticsDashboardButton />,
+    enabled: ({ isAnalyticsDashboard }) => isAnalyticsDashboard,
+  },
+  [BUTTON_CONFIG.EDIT_DASHBOARD]: {
+    component: ({ onRefreshPeriodChange }) => (
+      <EditDashboardButton onRefreshPeriodChange={onRefreshPeriodChange} />
+    ),
+    enabled: ({ isFullscreen, isEditing, canEdit }) =>
+      !isFullscreen && !isEditing && canEdit,
+  },
+  [BUTTON_CONFIG.DASHBOARD_SUBSCRIPTION]: {
+    component: () => <DashboardSubscriptionButton />,
+    enabled: ({
+      dashboard,
+      canManageSubscriptions,
+      formInput,
+      isAdmin,
+      isEditing,
+      isFullscreen,
+    }) =>
+      shouldRenderSubscriptionButton({
+        dashboard,
+        canManageSubscriptions,
+        formInput,
+        isAdmin,
+        isEditing,
+        isFullscreen,
+      }),
+  },
+  [BUTTON_CONFIG.DASHBOARD_EMBED_ACTION]: {
+    component: () => <DashboardEmbedAction />,
+    enabled: ({ dashboard, isPublic, isEmpty }) =>
+      !isEmpty && !isPublic && dashboard && !dashboard?.archived,
+  },
+  [BUTTON_CONFIG.REFRESH_WIDGET]: {
+    component: ({
+      refreshPeriod,
+      setRefreshElapsedHook,
+      onRefreshPeriodChange,
+    }) => (
+      <RefreshWidgetButton
+        period={refreshPeriod}
+        setRefreshElapsedHook={setRefreshElapsedHook}
+        onChangePeriod={onRefreshPeriodChange}
+      />
+    ),
+    enabled: ({ dashboard, isEmpty }) => !isEmpty && !dashboard?.archived,
```

Comment:
```suggestion
    enabled: ({ dashboard, isEmpty }) => !isEditing && !isEmpty && !dashboard?.archived,
```

The condition is missed, so is shows up in edit mode.
![image](https://github.com/user-attachments/assets/4884d7a1-a42f-46e6-8d56-8fbc9dd0c401)

Can you add tests to ensure this won't happen again? I think we have a bunch of E2E dashboard tests already setup, you could add a few assertions that expect the exact number of dashboard buttons and assert each of them or something like that.